plugin/src/main/java/com/microsoft/java/bs/gradle/plugin/SourceSetsModelBuilder.java

```diff
@@ -76,15 +83,24 @@ public Object buildAll(String modelName, Project rootProject) {
         gradleSourceSet.setSourceSetName(sourceSet.getName());
         gradleSourceSet.setClassesTaskName(sourceSet.getClassesTaskName());
 
-        // source
-        Set<File> srcDirs = sourceSet.getJava().getSrcDirs();
+        // source - gets Scala and Java sources
+        Set<File> srcDirs = sourceSet.getAllJava().getSrcDirs();
         gradleSourceSet.setSourceDirs(srcDirs);
         exclusionFromDependencies.addAll(srcDirs);
-        Set<File> generatedSrcDirs = new HashSet<>();
-        addAnnotationProcessingDir(project, sourceSet, generatedSrcDirs);
-        addGeneratedSourceDirs(project, sourceSet, srcDirs, generatedSrcDirs);
-        gradleSourceSet.setGeneratedSourceDirs(generatedSrcDirs);
-        exclusionFromDependencies.addAll(generatedSrcDirs);
+
+        // java
+        JavaCompile javaCompile = getJavaCompileTask(project, sourceSet);
+        if (javaCompile != null) {
+          gradleSourceSet.setIsJava(true);
+          gradleSourceSet.setSourceCompatibility(getSourceCompatibility(javaCompile));
+          gradleSourceSet.setTargetCompatibility(getTargetCompatibility(javaCompile));
+          gradleSourceSet.setJavaCompilerArgs(getJavaCompilerArgs(javaCompile));
+          Set<File> generatedSrcDirs = new HashSet<>();
+          addAnnotationProcessingDir(javaCompile, generatedSrcDirs);
```

Comment:
Does that mean Scala does not need dealing with the annotation processing?

Since scala plugin is extended from java plugin, when it comes to a project mixed with java and scala, is it possible that using annotation processing for the java part?

Reply:
Forget about it, I'm stupid. Since Scala plugin is extended from Java plugin, so JavaCompile task should be always available.